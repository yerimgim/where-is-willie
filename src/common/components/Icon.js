import React from "react";

import PropTypes from "prop-types";
import styled from "styled-components";

const Icon = ({ text, onClick, children, type }) => {
  return (
    <ButtonContainer type={type}>
      <Button onClick={onClick}>
        {children}
        <span>{text}</span>
      </Button>
    </ButtonContainer>
  );
};

const ButtonContainer = styled.div`
  span {
    font-size: 0.8rem;
    font-family: var(--noto-sans-kr-font);
    padding: 5px;
  }

  ${({ type }) => {
    return type === "close"
      ? `
      width: 100%;
      display: flex;
      justify-content: flex-end;
      font-size: 30px;
      padding: 10px 0;

      span {
        font-size: 1rem;
      }
    `
      : null;
  }}
`;

const Button = styled.div`
  display: inline-flex;
  align-items: center;
  margin: 0 5px;
  cursor: pointer;
`;

Icon.propTypes = {
  text: PropTypes.string,
  onClick: PropTypes.func,
  children: PropTypes.any,
  type: PropTypes.string,
};

export default Icon;
