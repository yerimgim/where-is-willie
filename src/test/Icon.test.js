import React from "react";

import { cleanup, fireEvent, render, screen } from "@testing-library/react";

import Icon from "../common/components/Icon";

afterEach(cleanup);

describe("<Icon />", () => {
  test("<Icon /> - getByText", () => {
    const text = "닫기";

    render(<Icon text={text} />);

    expect(screen.getByText(text)).toBeInTheDocument();
  });

  test("<Icon /> - onClick", () => {
    const handleClick = jest.fn();

    render(<Icon onClick={handleClick}>close</Icon>);

    fireEvent.click(screen.getByText(/close/i));
    expect(handleClick).toHaveBeenCalledTimes(1);
  });

  test("<Icon /> - type", () => {
    const { container } = render(<Icon variant="default" />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
