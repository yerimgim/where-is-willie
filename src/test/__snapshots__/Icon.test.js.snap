// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon /> <Icon /> - type 1`] = `
<div
  class="sc-bdvvtL iYxTvk"
>
  <div
    class="sc-gsDKAQ lbgLsI"
  >
    <span />
  </div>
</div>
`;
